src/polyhedralGravity/model/SplitNode.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <memory>
+
+#include "KdDefinitions.h"
+#include "TreeNode.h"
+
+namespace polyhedralGravity {
+
+    /**
+     * A TreeNode contained in a KDTree that splits the spatial hierarchy into two new sub boxes. Intersection tests are delegated to the child nodes.
+     */
+    class SplitNode final : public TreeNode {
+
+        /**
+     * The SplitNode that contains the bounding box closer to the origin with respect to the split plane
+     */
+        std::shared_ptr<TreeNode> _lesser;
```

Comment:
An often utilized practice is the utilization of `TreeNodePtr`.

For this to work, you require below the class definition something like
```c++
using TreeNodePtr = std::shared_ptr<TreeNode>;
```

I leave this for you, **as we currently don't use it anywhere in this project**. It's a matter of taste. So feel free to disregard this comment completely.